Fail to build latest version in OSX

Having this error when try to install 2.12.3 in Tahoe 26.4.1 with the latest Xcode

Ruby version: 3.3.7
Gem version: 3.5.22
Xcode version: 26.4 - Build version 17E192

```
current directory: /Users/XXXXXX/.local/share/mise/installs/ruby/3.3.7/lib/ruby/gems/3.3.0/gems/trilogy-2.12.3/ext/trilogy-ruby
/Users/XXXXXX/.local/share/mise/installs/ruby/3.3.7/bin/ruby extconf.rb
checking for whether -I /Users/XXXXXX/.local/share/mise/installs/ruby/3.3.7/lib/ruby/gems/3.3.0/gems/trilogy-2.12.3/ext/trilogy-ruby/inc is accepted as CFLAGS... no
checking for whether -std=gnu99 is accepted as CFLAGS... no
checking for whether -fvisibility=hidden is accepted as CFLAGS... no
checking for whether -DTRILOGY_XALLOCATOR is accepted as CFLAGS... no
checking for whether -g is accepted as CFLAGS... no
checking for CRYPTO_malloc() in -lcrypto... yes
checking for SSL_new() in -lssl... yes
checking for rb_ractor_local_storage_value_newkey() in ruby.h... yes
checking for rb_enc_interned_str() in ruby.h... yes
checking for rb_io_descriptor() in ruby.h... yes
creating Makefile

current directory: /Users/XXXXXX/.local/share/mise/installs/ruby/3.3.7/lib/ruby/gems/3.3.0/gems/trilogy-2.12.3/ext/trilogy-ruby
make DESTDIR\= sitearchdir\=./.gem.20260417-5863-sena0j sitelibdir\=./.gem.20260417-5863-sena0j clean

current directory: /Users/XXXXXX/.local/share/mise/installs/ruby/3.3.7/lib/ruby/gems/3.3.0/gems/trilogy-2.12.3/ext/trilogy-ruby
make DESTDIR\= sitearchdir\=./.gem.20260417-5863-sena0j sitelibdir\=./.gem.20260417-5863-sena0j
compiling trilogy.c
/Users/XXXXXX/.local/share/mise/installs/ruby/3.3.7/lib/ruby/gems/3.3.0/gems/trilogy-2.12.3/ext/trilogy-ruby/src/blocking.c:4:10: fatal error: 'trilogy/blocking.h' file
not found
    4 | #include "trilogy/blocking.h"
      |          ^~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [trilogy.o] Error 1

make failed, exit code 2
```

I fixed the error with this command:
```
gem install trilogy -v 2.12.3 -- --with-cflags="-I/Users/XXXXX/.local/share/mise/installs/ruby/3.3.7/lib/ruby/gems/3.3.0/gems/trilogy-2.12.3/ext/trilogy-ruby/inc" 2>&1
```

I also have ruby 3.4.8 installed in this machine and trilogy installed without issues as documented.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 
+### Fixed
+
+- Add workaround to fix compilation on older rubies combined in Xcode 26.4+.
+
 ## 2.12.3
 
 ### Fixed

--- a/contrib/ruby/ext/trilogy-ruby/extconf.rb
+++ b/contrib/ruby/ext/trilogy-ruby/extconf.rb
@@ -10,7 +10,17 @@ File.binwrite("trilogy.c",
   }.join)
 
 $objs = %w[trilogy.o cast.o cext.o]
-append_cflags(["-I #{__dir__}/inc", "-std=gnu99", "-fvisibility=hidden", "-DTRILOGY_XALLOCATOR", "-g"])
+append_cflags([
+  # Ref: https://github.com/trilogy-libraries/trilogy/issues/290
+  # Fix build on older ruby versions and Xcode 26.4+
+  "-Wno-default-const-init-field-unsafe",
+
+  "-I#{__dir__}/inc",
+  "-std=gnu99",
+  "-fvisibility=hidden",
+  "-DTRILOGY_XALLOCATOR",
+  "-g",
+])
 
 dir_config("openssl").any? || pkg_config("openssl")
 

